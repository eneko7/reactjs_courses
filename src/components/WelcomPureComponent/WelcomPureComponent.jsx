/* eslint-disable react/prefer-stateless-function */
import React, { PureComponent } from 'react';

class WelcomPureComponent extends PureComponent {
  render() {
    return (
      <div>Hello World - third method - React.PureComponent</div>
    );
  }
}

export default WelcomPureComponent;
