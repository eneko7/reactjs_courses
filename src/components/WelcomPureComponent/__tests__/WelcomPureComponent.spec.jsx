import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import WelcomPureComponent from '../WelcomPureComponent';

const renderer = new ShallowRenderer();
renderer.render(<WelcomPureComponent />);
describe('WelcomPureComponent', () => {
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
