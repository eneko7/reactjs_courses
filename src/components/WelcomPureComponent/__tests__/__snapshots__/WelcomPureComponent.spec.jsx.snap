// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomPureComponent renders correctly 1`] = `
<div>
  Hello World - third method - React.PureComponent
</div>
`;
